### PR TITLE
Announce exports

### PR DESCRIPTION
**Pull request by notspiff**
Let's try this again.. The potential for NIH and random ramblings should be minimal. I have zero patience with such behavior.

This adds announcements after a library export has finished. Requested here; http://forum.kodi.tv/showthread.php?tid=223761

### CODE REVIEW

**Comment by Memphiz** [comment]
Wait what who? April fools day? what? *shakes head, jumps circles* ... no brainer imo :) - this will fire after each xml file (if export to single files) or after the whole export (if export to one big xml file) - right?

**Comment by notspiff** [comment]
it fires once. if export is to a single file, the file exported is the payload of the announcement. if to multiple files, there is no payload.

**Comment by notspiff** [comment]
if you want to fire it once per file for singles, we can do that of course.

**Comment by Memphiz** [comment]
I couldn't care less - its more a question of what makes sense ;)

**Comment by notspiff** [comment]
my thoughts was that it would spam the bus, i.e. large impact little use.

**Comment by Memphiz** [comment]
Yep thats true. Fine.

**Comment by Montellese** [comment]
The general change looks good and I would keep it with sending a single announcement. Anything else will result in a lot of spam.
How big will a single export get? This might be a bit much data for remote running on very "weak" devices.
This also needs a change to `xbmc/interfaces/json-rpc/schema/notifications.json`.

**Comment by notspiff** [comment]
bad choice of words. the payload is the paths not the output.

**Comment by Montellese** [comment]
Ah yeah maybe choose easier to understand terms for `root` and `xml`.

PS: You could save the duplicate call code by passing the `CVariant` in either way but just keep it empty in case of single files.

**Comment by notspiff** [comment]
give your preferred names and i will change.

**Comment by uNiversaI** [comment]
I have a question I hope not to step on any toes

```ANNOUNCEMENT::CAnnouncementManager::Get().Announce(ANNOUNCEMENT::VideoLibrary, "xbmc", "OnExport");```

Why does it have to be xbmc why not @APP_NAME@ or appName? I asked similar in https://github.com/xbmc/xbmc/pull/6134 which seems to be going no where fast and deals with branding these exact announcement types.

**Comment by topfs2** [comment]
Its xbmc since its the caller, which was xbmc at the time. It has not been
changed yet and is not really part of this PR to do so.
Den 10 apr 2015 8:38 em skrev "uNiversaI" <notifications@github.com>:

**Comment by Montellese** [comment]
I'm not good with names either and I don't even know what the `root` property represents. Maybe simply use `file` or `exportedfile` instead of `xml`?

**Comment by notspiff** [comment]
nitpickery addressed.

**Comment by Montellese** [comment]
Thanks, looks good for me now. jenkins build this please

**Comment by Montellese** [comment]
Ah the changes to `xbmc/interfaces/json-rpc/schema/notifications.json` are still missing.

**Comment by notspiff** [comment]
grm, no idea how that shit works, is this okay?

**Comment by notspiff** [comment]
oh, and the 'root' property is the export path. thumbs etc are stuck in subfolders 'actors' 'movies' and so on.

File: xbmc/interfaces/json-rpc/schema/notifications.json [hunk 1]
```diff
@@ -192,6 +206,20 @@
...
     "returns": null
   },
+  "VideoLibrary.OnExport": {
+    "type": "notification",
+    "description": "A video library export has finished.",
+    "params": [
+      { "name": "sender", "type": "string", "required": true },
+      { "name": "data", "type": "object", "required": false,
+        "properties": {
+          "file": { "type": "string", "required": false},
```
Comment (Montellese) [inline xbmc/interfaces/json-rpc/schema/notifications.json]: This is missing the `root` property.

File: xbmc/interfaces/json-rpc/schema/notifications.json [hunk 2]
```diff
@@ -176,6 +176,20 @@
...
   },
+  "AudioLibrary.OnExport": {
+    "type": "notification",
+    "description": "An audio library export has finished.",
+    "params": [
+      { "name": "sender", "type": "string", "required": true },
+      { "name": "data", "type": "object", "required": false,
+        "properties": {
+          "file": { "type": "string", "required": false},
+          "failcount": { "type": "integer", "minimum": 0, "required": false}
```
Comment (Montellese) [inline xbmc/interfaces/json-rpc/schema/notifications.json]: You need to provide a default value for every property that is not required with e.g. `"default": 0`. Same goes for the `file` property.

**Comment by Montellese** [comment]
Almost yes :-)

**Comment by notspiff** [comment]
now, then ?

**Comment by Montellese** [comment]
Yup should be good. If you start xbmc and don't see any warning/error about JSON-RPC schema parsing you're good anyway. And http://jsonlint.com/ says it's valid JSON so all good.

**Comment by notspiff** [comment]
great, thanks for the handholding.

**Pull request closed without merging**